hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/endpoint/HeartbeatEndpointTask.java

```diff
@@ -147,8 +147,12 @@ public void setDatanodeDetailsProto(DatanodeDetailsProto
       rpcEndpoint.setLastSuccessfulHeartbeat(ZonedDateTime.now());
       rpcEndpoint.zeroMissedCount();
     } catch (IOException ex) {
-      // put back the reports which failed to be sent
-      putBackReports(requestBuilder);
+      // don't resend reports to recon as it could be down for days
+      // DN is expected to work fine without recon and not go OOM
```

Comment:
Why don't put back container actions and pipeline actions for SCM, or we have a legacy reason?

Reply:
@GlenGeng This change shouldn't affect SCM at all. Since SCM should be active (`rpcEndpoint.isPassive() == false`).

Reply:
Also I believe container actions and pipeline actions **are** put back on exception. cmiiw.

cc @nandakumar131  @lokeshj1703

Reply:
This is a concise fix. I am fine with it.
The question about container/pipeline Action won't be a blocker of merging this fix.